mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -363,4 +371,54 @@ class LogisticRegressionWithLBFGS
       new LogisticRegressionModel(weights, intercept, numFeatures, numOfLinearPredictor + 1)
     }
   }
+
+  /**
+   * Run the algorithm with the configured parameters on an input RDD
+   * of LabeledPoint entries starting from the initial weights provided.
+   * If a known updater is used calls the ml implementation, to avoid
+   * applying a regularization penalty to the intercept, otherwise
+   * defaults to the mllib implementation. If more than two classes
+   * or feature scaling is disabled, always uses mllib implementation.
+   */
+  override def run(input: RDD[LabeledPoint], initialWeights: Vector): LogisticRegressionModel = {
+    // ml's Logisitic regression only supports binary classifcation currently.
+    if (numOfLinearPredictor == 1 && useFeatureScaling) {
+      def runWithMlLogisitcRegression(elasticNetParam: Double) = {
+        // Prepare the ml LogisticRegression based on our settings
+        val lr = new org.apache.spark.ml.classification.LogisticRegression()
+        lr.setRegParam(optimizer.getRegParam())
+        lr.setElasticNetParam(elasticNetParam)
+        val initialWeightsWithIntercept = Vectors.dense(0.0, initialWeights.toArray:_*)
```

Comment:
Sorry, it will be a little messy. But when users don't set the weights in old api, we don't want to set the weights to the ml api since ml api has smarter way to compute the initialWeight. 

How about in `GeneralizedLinearAlgorithm`, have
```scala
private Boolean isUserSuppliedWeights = true
```
Then in ``` def run(input: RDD[LabeledPoint]): M = {``` 
```scala
isUserSuppliedWeights = false
```

Thus, we can have the following
```scala
if (isUserSuppliedWeights) {
  val initialWeightsWithIntercept = if(addIntercept) {
    appendBias(initialWeights)
  } else {
    initialWeights
  }
  lr.setInitialWeights(initialWeightsWithIntercept)
}
```